lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -689,15 +690,169 @@ def fill_in_high_snow_falling_levels(
+            gradient (numpy.array):
+                The gradient of the line of best fit we are using in the
+                extrapolation.
+            intercept (numpy.array):
+                The intercept of the line of best fit we are using in the
+                extrapolation.
+            snow_falling_level (numpy.array)
+                The snow falling level array with values filled in with snow
+                falling levels calculated through extrapolation.
+            sea points (numpy.array)
+                A boolean array with True where the points are sea points.
+
+        """
+
+        # Make sure we only try to extrapolate points with a valid gradient.
+        index = (gradient != 0.0) & sea_points
+        gradient = gradient[index]
+        intercept = intercept[index]
+        max_wb_int = max_wb_integral[index]
+        snow_fl = snow_falling_level[index]
+
+        # For points where -intercept/gradient is greater than zero:
+        index2 = (-intercept/gradient >= 0.0)
+        inside_sqrt = (
+            intercept[index2]**2 - 2*gradient[index2]*(
+                self.falling_level_threshold - max_wb_int[index2]))
+        snow_fl[index2] = (
+            (intercept[index2] - np.sqrt(inside_sqrt))/-gradient[index2])
+
+        # For points where -intercept/gradient is less than zero:
+        index2 = (-intercept/gradient < 0.0)
+        inside_sqrt = (
+            2*gradient[index2]*(
+                max_wb_int[index2] - self.falling_level_threshold))
+        snow_fl[index2] = (
+            (intercept[index2] - np.sqrt(inside_sqrt))/-gradient[index2])
+        # Update the snow falling level.
+        snow_falling_level[index] = snow_fl
+
+    @staticmethod
+    def linear_wet_bulb_fit(wet_bulb_temperature, heights, sea_points,
+                            start_point=0, end_point=5):
+        """
+        Calculates a linear fit to the wet bulb temperature profile close
+        to the surface to use when we extrapolate the wet bulb temperature
+        below sea level for sea points.
+
+        We only use a set number of points close to the surface for this fit,
+        specified by a start_point and end_point.
+
+        Args:
+            wet_bulb_temperature (numpy.array):
+                The wet bulb temperature profile at each grid point, with
+                height as the leading dimension.
+            heights (numpy.array):
+                The vertical height levels above orography, matching the
+                leading dimension of the wet_bulb_temperature.
+            sea_points (numpy.array):
+                A boolean array with True where the points are sea points.
+
+        Keyword Args:
+            start_point (int):
+                The index of the the starting height we want to use in our
+                linar fit.
+            end_point (int):
+                The index of the the end height we want to use in our
+                linar fit.
```

Comment:
Typo "linar"

Also, needs a return statement detailing the tuple that is passed back.

Reply:
Done